cluster-autoscaler/cloudprovider/aws/aws_manager.go

```diff
@@ -419,6 +449,36 @@ func (m *AwsManager) buildNodeFromTemplate(asg *asg, template *asgTemplate) (*ap
 	return &node, nil
 }
 
+func (m *AwsManager) getInstanceRequirementsFromMixedInstancesPolicy(policy *mixedInstancesPolicy) (*ec2.InstanceRequirements, error) {
+	instanceRequirements := &ec2.InstanceRequirements{}
+	if policy.launchTemplate != nil {
```

Comment:
a mixed instances policy requires an LT, so this should always be non-nil 

the check should be:

```
if policy.instanceRequirementsOverrides != nil {

} else if lt.LaunchTemplateData.InstanceRequirements != nil {

}
```

Basically if an LT override is specified in the MixedInstancesPolicy, then it should always be used (it overrides the LT parameter).
If there is no instance requirement override, then check if there are InstanceRequirements defined in the LT.